source/Calamari/Kubernetes/ResourceStatus/KubernetesYaml.cs

```diff
@@ -16,43 +18,43 @@ public static class KubernetesYaml
         /// Gets resource identifiers which are defined in a YAML file.
         /// A YAML file can define multiple resources, separated by '---'.
         /// </summary>
-        public static IEnumerable<ResourceIdentifier> GetDefinedResources(IEnumerable<string> manifests, string defaultNamespace)
+        public static IEnumerable<ResourceIdentifier> GetDefinedResources(IEnumerable<string> manifests,
+                                                                          IKubernetesManifestNamespaceResolver namespaceResolver,
+                                                                          IVariables variables,
+                                                                          ILog log)
         {
             foreach (var manifest in manifests)
             {
-                var input = new StringReader(manifest);
+                var yamlStream = new YamlStream();
+                yamlStream.Load(new StringReader(manifest));
```

Comment:
Just use the build-in `YamlStream` object, like the manifest reporter, for handling multiple documents in a single file.